packages/flutter/lib/src/widgets/widget_inspector.dart

```diff
@@ -1543,11 +1543,16 @@ mixin WidgetInspectorService {
   }
 
   bool _isLocalCreationLocation(_Location location) {
-    if (_pubRootDirectories == null || location == null || location.file == null) {
+    if (location == null || location.file == null) {
       return false;
     }
-
     final String file = Uri.parse(location.file).path;
+
+    // By default check whether the creation location was within package:flutter.
+    if (_pubRootDirectories == null) {
+      return !file.contains('packages/flutter/');
```

Comment:
I could imagine a scenario where I'm developing the Flutter part of a larger application, and we've used `packages/flutter` as a location in our project to include the Flutter part.

Is there some more robust way we can handle this?  Even something like `file.startsWith`?

Reply:
This is the easier way to do this until we may have symbolic link like pacage:flutter in the future. One idea I have is we can put the package root directory as a environment variable during flutter tool command, and read the variable during flutter test/run/build to register it to WidgetInspectorService, I am not sure how viable it is. thoughts? @jacob314 @jonahwilliams

Reply:
sadly file is currently an absolute path.
I'm working on switching it to a proper package: uri in the kernel transformer.
When this lands we will be able to switch this code to
uri.startsWith('package:flutter/')

Reply:
do you have the issue id I can refer to?

Reply:
https://github.com/flutter/flutter/issues/32660

Reply:
added a todo comment